modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/RemoveProcessor.java

```diff
@@ -57,19 +59,11 @@ public String getType() {
 
     public static final class Factory implements Processor.Factory {
 
-        private final TemplateService templateService;
-
-        public Factory(TemplateService templateService) {
-            this.templateService = templateService;
-        }
-
         @Override
         public RemoveProcessor create(Map<String, Processor.Factory> registry, String processorTag,
                                       Map<String, Object> config) throws Exception {
-            String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
-            TemplateService.Template compiledTemplate = ConfigurationUtils.compileTemplate(TYPE, processorTag,
-                "field", field, templateService);
-            return new RemoveProcessor(processorTag, compiledTemplate);
+            List<String> fields = ConfigurationUtils.readList(TYPE, processorTag, config, "field");
```

Comment:
I think it would be nice to support both singular strings *and* lists. what do you think? Will also help with backwards compatibility